does not compile under 5.1.0~alpha2

```bash
# make
dune build @all
File "src/batMarshal.mli", lines 59-62, characters 0-13:
Error: This variant or record definition does not match that of type
         Marshal.extern_flags
       An extra constructor, Compression, is provided in the original definition.
make: *** [Makefile:4: all] Error 1        
```

--- a/src/batArray.mli
+++ b/src/batArray.mli
@@ -928,4 +928,10 @@ external unsafe_set : 'a array -> int -> 'a -> unit = "%array_unsafe_set"
 ##V>=4.6##      = "%floatarray_unsafe_set"
 ##V>=4.6##end
 
+##V>=5.1##val find_index : ('a -> bool) -> 'a array -> int option
+##V>=5.1##val find_map : ('a -> 'b option) -> 'a array -> 'b option
+##V>=5.1##val find_mapi : (int -> 'a -> 'b option) -> 'a array -> 'b option
+##V>=5.1##val map_inplace : ('a -> 'a) -> 'a array -> unit
+##V>=5.1##val mapi_inplace : (int -> 'a -> 'a) -> 'a array -> unit
+
 (**/**)

--- a/src/batChar.mli
+++ b/src/batChar.mli
@@ -150,6 +150,8 @@ val compare: t -> t -> int
 val equal : t -> t -> bool
 val hash : t -> int
 
+##V>=5.1##val seeded_hash : int -> t -> int
+
 val ord : char BatOrd.ord
 
 module Incubator : sig

--- a/src/batInt32.ml
+++ b/src/batInt32.ml
@@ -195,3 +195,6 @@ let min (x: t) (y: t): t =
 
 let max (x: t) (y: t): t =
   if x >= y then x else y
+
+##V>=5.1##let seeded_hash = Int32.seeded_hash
+##V>=5.1##let hash = Int32.hash

--- a/src/batInt32.mli
+++ b/src/batInt32.mli
@@ -325,3 +325,6 @@ val print: 'a BatInnerIO.output -> t -> unit
 
 val print_hex: 'a BatInnerIO.output -> t -> unit
   (** prints as hex string *)
+
+##V>=5.1##val seeded_hash : int -> t -> int
+##V>=5.1##val hash : t -> int

--- a/src/batInt64.ml
+++ b/src/batInt64.ml
@@ -77,3 +77,6 @@ let min (x: t) (y: t): t =
 
 let max (x: t) (y: t): t =
   if x >= y then x else y
+
+##V>=5.1##let seeded_hash = Int64.seeded_hash
+##V>=5.1##let hash = Int64.hash

--- a/src/batInt64.mli
+++ b/src/batInt64.mli
@@ -304,3 +304,6 @@ val print: 'a BatInnerIO.output -> t -> unit
 
 val print_hex: 'a BatInnerIO.output -> t -> unit
   (** prints as hex string *)
+
+##V>=5.1##val seeded_hash : int -> t -> int
+##V>=5.1##val hash : t -> int

--- a/src/batList.ml
+++ b/src/batList.ml
@@ -64,6 +64,8 @@ let rev_map2 = List.rev_map2
 ##V>=4.10##let concat_map = List.concat_map
 ##V>=4.10##let find_map_opt = List.find_map
 ##V>=4.12##let equal = List.equal
+##V>=5.1##let find_index = List.find_index
+##V>=5.1##let find_mapi = List.find_mapi
 
 (* ::VH:: END GLUE *)
 

--- a/src/batList.mli
+++ b/src/batList.mli
@@ -1169,3 +1169,6 @@ end
 
 val ( @ ) : 'a list -> 'a list -> 'a list
   (** Tail recursive [List.append]. *)
+
+##V>=5.1##val find_index : ('a -> bool) -> 'a list -> int option
+##V>=5.1##val find_mapi : (int -> 'a -> 'b option) -> 'a list -> 'b option

--- a/src/batMap.ml
+++ b/src/batMap.ml
@@ -1009,6 +1009,9 @@ sig
   val to_seq_from :  key -> 'a t -> (key * 'a) BatSeq.t
   val add_seq : (key * 'a) BatSeq.t -> 'a t -> 'a t
   val of_seq : (key * 'a) BatSeq.t -> 'a t
+  val to_list : 'a t -> (key * 'a) list
+  val of_list : (key * 'a) list -> 'a t
+  val add_to_list: key -> 'a -> 'a list t -> 'a list t
 
   (** {7 Printing}*)
 
@@ -1163,6 +1166,13 @@ struct
   let to_seq m = Concrete.to_seq (impl_of_t m)
   let to_rev_seq m = Concrete.to_rev_seq (impl_of_t m)
   let to_seq_from k m = Concrete.to_seq_from Ord.compare k (impl_of_t m)
+
+  let add_to_list x data m =
+    let add = function None -> Some [data] | Some l -> Some (data :: l) in
+    update_stdlib x add m
+
+  let to_list = bindings
+  let of_list bs = List.fold_left (fun m (k, v) -> add k v m) empty bs
 
   module Exceptionless =
   struct

--- a/src/batMap.mli
+++ b/src/batMap.mli
@@ -446,6 +446,10 @@ sig
     
       @since 3.3.0 *)
 
+  val to_list : 'a t -> (key * 'a) list
+  val of_list : (key * 'a) list -> 'a t
+  val add_to_list: key -> 'a -> 'a list t -> 'a list t
+
   (** {6 Boilerplate code}*)
 
   (** {7 Printing}*)

--- a/src/batMarshal.mli
+++ b/src/batMarshal.mli
@@ -60,6 +60,7 @@ type extern_flags = Marshal.extern_flags =
     No_sharing                          (** Don't preserve sharing *)
   | Closures                            (** Send function closures *)
 ##V>=4.1##  | Compat_32                           (** Ensure 32-bit compatibility *)
+##V>=5.1##  | Compression                         (** Compress the output if possible *)
 (** The flags to the [Marshal.to_*] functions below. *)
 
 
@@ -99,6 +100,8 @@ val output: _ BatInnerIO.output -> ?sharing:bool -> ?closures:bool -> 'a -> unit
 
    @since 2.3.0
 *)
+
+##V>=5.1##val compression_supported : unit -> bool
 
 external to_string :
   'a -> extern_flags list -> string = "caml_output_value_to_string"

--- a/src/batNativeint.ml
+++ b/src/batNativeint.ml
@@ -82,6 +82,9 @@ external format : string -> nativeint -> string = "caml_nativeint_format"
 ##V>=4.08##let unsigned_rem = Nativeint.unsigned_rem
 ##V>=4.08##let unsigned_div = Nativeint.unsigned_div
 
+##V>=5.1##let seeded_hash = Nativeint.seeded_hash
+##V>=5.1##let hash = Nativeint.hash
+
 type bounded = t
 let min_num, max_num = min_int, max_int
 

--- a/src/batNativeint.mli
+++ b/src/batNativeint.mli
@@ -252,6 +252,9 @@ val max: t -> t -> t
 ##V>=4.08##
 ##V>=4.08##    @since 2.10.0 and OCaml 4.08.0 *)
 
+##V>=5.1##val seeded_hash : int -> t -> int
+##V>=5.1##val hash : t -> int
+
 val equal : t -> t -> bool
 (** Equality function for 64-bit integers, useful for {!HashedType}. *)
 

--- a/src/batRandom.ml
+++ b/src/batRandom.ml
@@ -68,7 +68,7 @@ struct
   let enum_bool state () = BatEnum.from (fun () -> bool state)
   let enum_char state () = BatEnum.from (fun () -> char state)
 
-##V>=5##  external next: t -> (int64[@unboxed]) 
+##V>=5##  external next: t -> (int64[@unboxed])
 ##V>=5##    = "caml_lxm_next" "caml_lxm_next_unboxed" [@@noalloc]
   
 end

--- a/src/batRandom.mli
+++ b/src/batRandom.mli
@@ -216,6 +216,9 @@ module State : sig
 ##V>=5##    correlation.  Both PRNGs can be split later, arbitrarily many times.
 ##V>=5##    @since 3.6.0 and OCaml 5.0.0 *)
 
+##V>=5.1##val to_binary_string : t -> string
+##V>=5.1##val of_binary_string : string -> t
+
 end
 
 

--- a/src/batSeq.ml
+++ b/src/batSeq.ml
@@ -458,6 +458,8 @@ let of_string ?(first="[") ?(last="]") ?(sep=";") of_str s =
 ##V>=4.14##let to_dispenser           = Stdlib.Seq.to_dispenser
 ##V>=4.14##let ints                   = Stdlib.Seq.ints
 ##V>=4.14##let equal_stdlib           = Stdlib.Seq.equal
+##V>=5.1##let find_index = Stdlib.Seq.find_index
+##V>=5.1##let find_mapi = Stdlib.Seq.find_mapi
 
 module Infix = struct
   (** Infix operators matching those provided by {!BatEnum.Infix} *)

--- a/src/batSeq.mli
+++ b/src/batSeq.mli
@@ -886,3 +886,6 @@ end
 ##V>=4.14##
 ##V>=4.14##    @since 4.14 *)
 ##V>=4.14##
+
+##V>=5.1##val find_index : ('a -> bool) -> 'a t -> int option
+##V>=5.1##val find_mapi : (int -> 'a -> 'b option) -> 'a t -> 'b option

--- a/src/batSplay.ml
+++ b/src/batSplay.ml
@@ -670,6 +670,9 @@ struct
 
   let to_list m = List.of_enum (enum m)
   let of_list l = of_enum (List.enum l)
+  let add_to_list x data m =
+    let add = function None -> Some [data] | Some l -> Some (data :: l) in
+    update_stdlib x add m
 
   let custom_print ~first ~last ~sep kvpr out m =
     Enum.print ~first ~last ~sep

--- a/src/batSys.mli
+++ b/src/batSys.mli
@@ -169,6 +169,8 @@ val max_array_length : int
 ##V>=4.3##    as the contents of the [OCAMLRUNPARAM] environment variable.
 ##V>=4.3##    @since 2.5.0 and OCaml 4.03.0 *)
 
+##V>=5.1##external is_regular_file : string -> bool = "caml_sys_is_regular_file"
+
 (** {6 Signal handling} *)
 
 

